Add value checking on relabel command for selinux

rhatdan opened the pull request:
Currently relabel requires an exact match on passed in relabel string.  This patch will allow the passing of multiple params in the string, and just look for Z or z

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

mrunalp commented on label/label_selinux.go:
```diff
@@ -104,7 +104,13 @@ func Relabel(path string, fileLabel string, relabel string) error {
+	if !strings.ContainsAny(relabel, "z & Z") {
+		return nil
+	}
+	if strings.Contains(relabel, "z & Z") {
```
I think this is probably incorrect, unless you are searching for the substring "z & Z" in relabel.

rhatdan replied:
Yes I think you are right, I am looking for ContainersBoth

mrunalp commented:
You have a merge commit. Could you rebase?

mrunalp commented on label/label_selinux.go:
```diff
@@ -104,7 +104,13 @@ func Relabel(path string, fileLabel string, relabel string) error {
 		return nil
 	}
-	if relabel == "z" {
+	if !strings.ContainsAny(relabel, "z & Z") {
```
You can do this here otherwise getting a space or & will also satisfy the condition.
```go
if !strings.ContainsAny(relabel, "zZ") {
```

rhatdan commented:
@mrunalp Must have had my head up my butt when doing the patch originally.  I googled for an example, and this is what I saw.   Fixed.  I also added a small patch to fix the label on /dev/mqueue.  I probably should keep these as two different pull requests.

mrunalp commented:
@rhatdan Ahh, okay :) Yes, you should create a separate PR for the mqueue fix.

mrunalp commented on label/label_selinux.go:
```diff
@@ -104,7 +104,13 @@ func Relabel(path string, fileLabel string, relabel string) error {
+	if !strings.ContainsAny(relabel, "zZ") {
+		return nil
+	}
+	if strings.Contains(relabel, "z") {
```
Might be better to do:
```
                if strings.Contains(relabel, "z") && strings.Contains(relabel, "Z") {
		            return fmt.Errorf("Bad SELinux Option z & Z can not be used together")
                }
                c := selinux.NewContext(fileLabel)
                c["level"] = "s0"
                fileLabel = c.Get()
                return selinux.Chcon(path, fileLabel, true)
```

rhatdan commented:
Back to the original way I had it, and split out the mqueue fix to a separate patch.

mrunalp commented:
@rhatdan Thanks! LGTM

mrunalp commented:
@dqminh Could you take a look?

dqminh commented:
LGTM
